### PR TITLE
[PLAT-2437] Add executeAborted event for box-query tool

### PR DESCRIPTION
**Pull request by jdm717**
## Summary

Adds an `executeAborted` event dispatcher to the volume intersection controller in the case that we receive too many items that match the query. This can then be handled on the consumer-side rather than logging an error for the expected behavior.

## Test Plan

- Verify that there is no console.error log for queries that are aborted
- Verify the listener registered using the `onExecuteAborted` method is called with the error

## Release Notes

N/A

## Possible Regressions

Volume intersection query error handling

## Dependencies

N/A

### CODE REVIEW

File: packages/viewer/src/lib/volume-intersection/controller.ts [hunk 1]
```diff
@@ -108,8 +114,15 @@ export class VolumeIntersectionQueryController {
           ])
           .execute();
       } catch (e) {
-        console.error('Failed to perform volume intersection query', e);
-        throw e;
+        if (
+          e instanceof StreamRequestError &&
+          e.summary?.toLocaleLowerCase().includes('operation aborted')
```
Comment (jdm717) [inline packages/viewer/src/lib/volume-intersection/controller.ts:119]: May be a good tech-talk topic, but we don't have any other context around this error message currently, so the summary is the only way to determine this is intentional. Should we be sending more information with these request errors from FSS? I.e. should these messages also include codes that we can leverage?

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx [hunk 2]
```diff
@@ -252,7 +252,7 @@ export class ViewerBoxQueryTool {
   private setDefaultClearAndSelectOperation(): void {
     this.controller?.setOperationTransform((builder) => builder.select());
     this.controller?.setAdditionalTransforms([
-      (op) => op.where((q) => q.all()).deselect(),
+      (op) => op.where((q) => q.withSelected()).deselect(),
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:255]: We should figure out why this change is necessary.

File: packages/viewer/src/lib/volume-intersection/controller.ts (hunk 1 above)
Comment (MadisonEhlers-Vertex) [reply packages/viewer/src/lib/volume-intersection/controller.ts:126]: I agree, we should be using status codes, and sending those back to the client. Worth a discussion and maybe a ticket.

**Review by MadisonEhlers-Vertex — APPROVED** [review]

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx (hunk 2 above)
Comment (jdm717) [reply packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:255]: Agreed, I created https://vertexvis.atlassian.net/browse/PLAT-2506 to capture this issue
